master/internal/db/postgres.go

```diff
@@ -1119,13 +1119,37 @@ WHERE id = :id`, setClause(toUpdate)), trial)
 	return nil
 }
 
+// RollbackSearcherEvents rolls back the events for an experiment to the last step with a
+// checkpoint.
+func (db *PgDB) RollbackSearcherEvents(experimentID int) error {
+	_, err := db.sql.Exec(`
+DELETE FROM searcher_events se
+USING (
+    SELECT
+        (se.content->'msg'->'workload'->>'trial_id')::int trial_id,
+        max(CASE WHEN content->'msg'->'workload'->>'kind' = 'CHECKPOINT_MODEL'
+        THEN (se.content->'msg'->'workload'->>'step_id')::int ELSE 0 END) latest_checkpoint_id
+    FROM searcher_events se
+    GROUP BY trial_id
+	) lc
+WHERE experiment_id = $1
+    AND (se.content->'msg'->'workload'->>'trial_id')::int = lc.trial_id
+    AND (se.content->'msg'->'workload'->>'step_id')::int > lc.latest_checkpoint_id;
+	`, experimentID)
```

Comment:
Miss this by accident because it was hidden in the middle (damn github, I dont like that feature). I'll leave a good comment. Query surprisingly isn't terrible performance wise. I was trying instead for something more like 
```
DELETE FROM searcher_events se
USING (
    SELECT
        (se.content->'msg'->'workload'->>'trial_id')::int trial_id,
        first(content->'msg'->'workload'->>'kind' = 'CHECKPOINT_MODEL')::int step_id
    FROM searcher_events se
	GROUP BY trial_id
	ORDER BY (se.content->'msg'->'workload'->>'step_id')::int DESC
	) latest_checkpoint
WHERE experiment_id = 1
    AND (se.content->'msg'->'workload'->>'trial_id')::int = lclatest_checkpoint.trial_id
    AND (se.content->'msg'->'workload'->>'step_id')::int > latest_checkpoint.step_id;
```
with an index like
```
CREATE INDEX ix_searcher_events_exp_trial_step_id ON public.searcher_events USING btree (
  experiment_id,
  ((searcher_events.content->'msg'->'workload'->>'trial_id')::int),
  ((searcher_events.content->'msg'->'workload'->>'step_id')::int)
);
```
but I can't quite get it to work (postgres complaining about bad usage of `first`, I guess I dont know how the first agg function works).